src/code/z_parameter.c

```diff
@@ -1271,7 +1276,7 @@ void Inventory_SwapAgeEquipment(void) {
     }
 
     shieldEquipValue = gEquipMasks[EQUIP_TYPE_SHIELD] & gSaveContext.save.info.equips.equipment;
-    if (shieldEquipValue != 0) {
+    if ((u32)shieldEquipValue != 0) {
         shieldEquipValue >>= gEquipShifts[EQUIP_TYPE_SHIELD];
```

Comment:
this is needed to match?

Reply:
Maybe, there might be a different solution but retail had a a1-a2 register swap and doing `if (shieldEquipValue) {` fixes it, and using the `(u32)` is an artificial hack to keep the 0. 

![image](https://github.com/zeldaret/oot/assets/47598039/a68e6923-1af6-44d2-801b-72d410cf0db8)

Reply:
oh right, its the hack to keep the comparison. okay, `if (shieldEquipValue)` makes more sense for what was probably written

Reply:
Yeah, it does make more sense. Changed

Reply:
oh I wasnt saying we need to change it-- we've already started consistently using the hack in other places iirc. I was never a fan of using it to begin with but its what we decided on iirc

Reply:
I don't think we've been putting the `!= 0` for flags and bitchecks though right? Since
```c
shieldEquipValue = gEquipMasks[EQUIP_TYPE_SHIELD] & gSaveContext.save.info.equips.equipment;
if (shieldEquipValue) {
```
is really just checking 
```c
if (gEquipMasks[EQUIP_TYPE_SHIELD] & gSaveContext.save.info.equips.equipment)
```

i.e. would we have
```c
if ((gEquipMasks[EQUIP_TYPE_SHIELD] & gSaveContext.save.info.equips.equipment) != 0)
```

Reply:
oh, right okay